### PR TITLE
feat: Add action to delete Advisories.

### PR DESCRIPTION
**Pull request by chirino**
A task of https://github.com/trustification/trustify/issues/563

### CODE REVIEW

**Review by gildub — CHANGES_REQUESTED** [review]
Same issue here in regards of not using React Query layer for API requests.
Please update accordingly.

File: client/src/app/queries/advisories.ts [hunk 1]
```diff
@@ -56,6 +59,24 @@ export const useFetchAdvisoryById = (id?: number | string) => {
...
 };
 
+export const useDeleteAdvisoryByIdMutation = () => {
+  const queryClient = useQueryClient();
+  const {pushNotification} = React.useContext(NotificationsContext);
+
+  return useMutation({
+    mutationFn: (id: number | string) => deleteAdvisoryById(id),
+    onSuccess: (_res, id) => {
+      queryClient.invalidateQueries({queryKey: [AdvisoriesQueryKey, id]});
```
Comment (gildub) [inline client/src/app/queries/advisories.ts:69]: The invalidateQueries key could actually be :
`queryKey: [SBOMsQueryKey]`

**Review by gildub — CHANGES_REQUESTED** [review]
LGTM just a minor change, please see inline

**Review by carlosthe19916 — CHANGES_REQUESTED** [review]
Thanks for the PR. The same comments at https://github.com/trustification/trustify-ui/pull/120#pullrequestreview-2204769060 apply here

**Comment by chirino** [comment]
rolled into https://github.com/trustification/trustify-ui/pull/129

**Pull request closed without merging**